src/Model/CustomerTokenService.php

```diff
@@ -106,16 +115,21 @@ public function __construct(
      * @param $password
      * @return string
      */
-    public function createCustomerAccessToken($username, $password): string
+    public function createCustomerAccessToken($username, $password, $storeId = null): string
     {
         $this->validatorHelper->validate($username, $password);
         $this->getRequestThrottler()->throttle($username, RequestThrottler::USER_TYPE_CUSTOMER);
 
         try {
             $customerDataObject = $this->accountManagement->authenticate($username, $password);
         } catch (EmailNotConfirmedException $e) {
+            $href = $this->urlBuilder->getUrl(
+                'customer/account/confirmation',
+                ['_query' => ['store' => $storeId, 'email' => $username]]
```

Comment:
not 100% certain, but you may want to check if $storeId is not null here

Reply:
Do I need to add separate requests (with and without store) if it returns the correct URL even if storeId=null(in this case, the address for the default store is returned)?

Reply:
if it works fine with null, then never-mind, can keep as is